### PR TITLE
Spec test: capella and update to `v1.3.0-alpha.1`

### PR DESCRIPTION
**Pull request by terencechain**
This PR adds Capella spec tests and updates spec tests to the latest alpha release 

Pre-req to merging this PR:
- [x] `ProcessBLSToExecutionChanges` changes in `beacon-chain/core/blocks/withdrawals.go`
- [x] State transition change in `beacon-chain/core/transition/transition_no_verify_sig.go`
- [x] Various bug fixes under `beacon-chain/state`

### CODE REVIEW

File: config/fieldparams/mainnet.go [hunk 1]
```diff
@@ -26,6 +26,4 @@ const (
 	SyncCommitteeAggregationBytesLength   = 16            // SyncCommitteeAggregationBytesLength defines the length of sync committee aggregate bytes.
 	SyncAggregateSyncCommitteeBytesLength = 64            // SyncAggregateSyncCommitteeBytesLength defines the length of sync committee bytes in a sync aggregate.
 	MaxWithdrawalsPerPayload              = 16            // MaxWithdrawalsPerPayloadLength defines the maximum number of withdrawals that can be included in a payload.
-	WithdrawalQueueLimit                  = 1099511627776 // WithdrawalQueueLimit defines the maximum number of withdrawals queued in the state.
-	ExecutionAddressLength                = 20            // ExecutionAddressLength defines the length of an execution layer address.
```
Comment (prestonvanloon) [inline config/fieldparams/mainnet.go:30]: Why are these deleted?

File: config/fieldparams/minimal.go [hunk 2]
```diff
@@ -25,6 +25,5 @@ const (
 	SlotsPerEpoch                         = 8             // SlotsPerEpoch defines the number of slots per epoch.
 	SyncCommitteeAggregationBytesLength   = 1             // SyncCommitteeAggregationBytesLength defines the sync committee aggregate bytes.
 	SyncAggregateSyncCommitteeBytesLength = 4             // SyncAggregateSyncCommitteeBytesLength defines the length of sync committee bytes in a sync aggregate.
-	MaxWithdrawalsPerPayload              = 16            // MaxWithdrawalsPerPayloadLength defines the maximum number of withdrawals that can be included in a payload.
-	WithdrawalQueueLimit                  = 1099511627776 // WithdrawalQueueLimit defines the maximum number of withdrawals queued in the state.
+	MaxWithdrawalsPerPayload              = 4             // MaxWithdrawalsPerPayloadLength defines the maximum number of withdrawals that can be included in a payload.
```
Comment (prestonvanloon) [inline config/fieldparams/minimal.go:28]: Same here, why these changes to presents?

File: testing/spectest/mainnet/capella/epoch_processing/BUILD.bazel [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+        "inactivity_updates_test.go",
+        "justification_and_finalization_test.go",
+        "participation_flag_updates_test.go",
+        "randao_mixes_reset_test.go",
+        "registry_updates_test.go",
+        "rewards_and_penalties_test.go",
+        "slashings_reset_test.go",
+        "slashings_test.go",
+    ],
+    data = glob(["*.yaml"]) + [
```
Comment (prestonvanloon) [inline testing/spectest/mainnet/capella/epoch_processing/BUILD.bazel:19]:
```suggestion
    data = [
```
There are no yaml files here that would match that glob.

File: testing/spectest/mainnet/capella/finality/BUILD.bazel [hunk 4]
```diff
@@ -0,0 +1,14 @@
+load("@prysm//tools/go:def.bzl", "go_test")
+
+go_test(
+    name = "go_default_test",
+    size = "medium",
+    timeout = "short",
+    srcs = ["finality_test.go"],
+    data = glob(["*.yaml"]) + [
```
Comment (prestonvanloon) [inline testing/spectest/mainnet/capella/finality/BUILD.bazel:8]:
```suggestion
    data = [
```
Same here, there are no local yaml files that would match this glob

File: testing/spectest/mainnet/capella/finality/BUILD.bazel [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+        "@consensus_spec_tests_mainnet//:test_data",
+    ],
+    shard_count = 4,
```
Comment (prestonvanloon) [inline testing/spectest/mainnet/capella/finality/BUILD.bazel:11]: is this shard count appropriate? This looks like 1 test only.

```suggestion
    shard_count = 1,
```

File: testing/spectest/mainnet/capella/fork_helper/BUILD.bazel [hunk 6]
```diff
@@ -0,0 +1,13 @@
+load("@prysm//tools/go:def.bzl", "go_test")
+
+go_test(
+    name = "go_default_test",
+    size = "small",
+    srcs = ["upgrade_to_capella_test.go"],
+    data = glob(["*.yaml"]) + [
```
Comment (prestonvanloon) [inline testing/spectest/mainnet/capella/fork_helper/BUILD.bazel:7]:
```suggestion
    data = [
```

File: testing/spectest/mainnet/capella/fork_helper/BUILD.bazel [hunk 7]
```diff
@@ -0,0 +1,13 @@
...
+        "@consensus_spec_tests_mainnet//:test_data",
+    ],
+    shard_count = 4,
```
Comment (prestonvanloon) [inline testing/spectest/mainnet/capella/fork_helper/BUILD.bazel:10]:
```suggestion
    shard_count = 1,
```

File: testing/spectest/mainnet/capella/fork_transition/BUILD.bazel [hunk 8]
```diff
@@ -0,0 +1,12 @@
+load("@prysm//tools/go:def.bzl", "go_test")
+
+go_test(
+    name = "go_default_test",
+    size = "small",
+    srcs = ["transition_test.go"],
+    data = glob(["*.yaml"]) + [
```
Comment (prestonvanloon) [inline testing/spectest/mainnet/capella/fork_transition/BUILD.bazel:7]:
```suggestion
    data = [
```

File: testing/spectest/mainnet/capella/forkchoice/BUILD.bazel [hunk 9]
```diff
@@ -0,0 +1,17 @@
+load("@prysm//tools/go:def.bzl", "go_test")
+
+go_test(
+    name = "go_default_test",
+    size = "enormous",
+    timeout = "short",
+    srcs = ["forkchoice_test.go"],
+    data = glob(["*.yaml"]) + [
```
Comment (prestonvanloon) [inline testing/spectest/mainnet/capella/forkchoice/BUILD.bazel:8]:
```suggestion
    data = [
```

File: testing/spectest/mainnet/capella/operations/BUILD.bazel [hunk 10]
```diff
@@ -0,0 +1,21 @@
...
+    srcs = [
+        "attestation_test.go",
+        "attester_slashing_test.go",
+        "block_header_test.go",
+        "deposit_test.go",
+        "proposer_slashing_test.go",
+        "sync_committee_test.go",
+        "voluntary_exit_test.go",
+    ],
+    data = glob(["*.yaml"]) + [
```
Comment (prestonvanloon) [inline testing/spectest/mainnet/capella/operations/BUILD.bazel:15]:
```suggestion
    data = [
```

File: testing/spectest/mainnet/capella/sanity/BUILD.bazel [hunk 11]
```diff
@@ -0,0 +1,16 @@
...
+
+go_test(
+    name = "go_default_test",
+    size = "medium",
+    timeout = "short",
+    srcs = [
+        "blocks_test.go",
+        "slots_test.go",
+    ],
+    data = glob(["*.yaml"]) + [
```
Comment (prestonvanloon) [inline testing/spectest/mainnet/capella/sanity/BUILD.bazel:11]:
```suggestion
    data = [
```

File: testing/spectest/minimal/capella/forkchoice/BUILD.bazel [hunk 12]
```diff
@@ -0,0 +1,21 @@
+load("@prysm//tools/go:def.bzl", "go_test")
+
+go_test(
+    name = "go_default_test",
+    size = "enormous",
+    timeout = "short",
+    srcs = ["forkchoice_test.go"],
+    data = glob(["*.yaml"]) + [
```
Comment (prestonvanloon) [inline testing/spectest/minimal/capella/forkchoice/BUILD.bazel:8]:
```suggestion
    data = [
```

File: testing/spectest/minimal/capella/operations/BUILD.bazel [hunk 13]
```diff
@@ -0,0 +1,26 @@
...
+        "attester_slashing_test.go",
+        "block_header_test.go",
+        "bls_to_execution_change_test.go",
+        "deposit_test.go",
+        "proposer_slashing_test.go",
+        "sync_committee_test.go",
+        "voluntary_exit_test.go",
+        "withdrawals_test.go",
+    ],
+    data = glob(["*.yaml"]) + [
```
Comment (prestonvanloon) [inline testing/spectest/minimal/capella/operations/BUILD.bazel:17]:
```suggestion
    data = [
```

File: testing/spectest/minimal/capella/rewards/BUILD.bazel [hunk 14]
```diff
@@ -0,0 +1,16 @@
+load("@prysm//tools/go:def.bzl", "go_test")
+
+go_test(
+    name = "go_default_test",
+    size = "small",
+    srcs = ["rewards_test.go"],
+    data = glob(["*.yaml"]) + [
```
Comment (prestonvanloon) [inline testing/spectest/minimal/capella/rewards/BUILD.bazel:7]:
```suggestion
    data = [
```

File: testing/spectest/minimal/capella/sanity/BUILD.bazel [hunk 15]
```diff
@@ -0,0 +1,19 @@
+load("@prysm//tools/go:def.bzl", "go_test")
+
+go_test(
+    name = "go_default_test",
+    size = "small",
+    srcs = [
+        "blocks_test.go",
+        "slots_test.go",
+    ],
+    data = glob(["*.yaml"]) + [
```
Comment (prestonvanloon) [inline testing/spectest/minimal/capella/sanity/BUILD.bazel:10]:
```suggestion
    data = [
```

File: testing/spectest/minimal/capella/ssz_static/BUILD.bazel [hunk 16]
```diff
@@ -0,0 +1,16 @@
+load("@prysm//tools/go:def.bzl", "go_test")
+
+go_test(
+    name = "go_default_test",
+    size = "small",
+    srcs = ["ssz_static_test.go"],
+    data = glob(["*.yaml"]) + [
```
Comment (prestonvanloon) [inline testing/spectest/minimal/capella/ssz_static/BUILD.bazel:7]:
```suggestion
    data = [
```

**Review by prestonvanloon — COMMENTED** [review]
Tests fail

File: config/fieldparams/mainnet.go (hunk 1 above)
Comment (terencechain) [reply config/fieldparams/mainnet.go:30]: They are not used, and they will never be used so removed unused variables

File: config/fieldparams/minimal.go (hunk 2 above)
Comment (terencechain) [reply config/fieldparams/minimal.go:28]: The correct value for `MaxWithdrawalsPerPayload` is [4](https://github.com/ethereum/consensus-specs/blob/dev/presets/minimal/capella.yaml)
This has to be fixed or else the minimal spec test will fail

File: consensus-types/blocks/execution.go [hunk 17]
```diff
@@ -647,10 +647,6 @@ func PayloadToHeaderCapella(payload interfaces.ExecutionData) (*enginev1.Executi
 // IsEmptyExecutionData checks if an execution data is empty underneath. If a single field has
 // a non-zero value, this function will return false.
 func IsEmptyExecutionData(data interfaces.ExecutionData) (bool, error) {
-	_, ok := data.Proto().(*enginev1.ExecutionPayloadCapella)
-	if ok {
-		return false, nil
-	}
```
Comment (terencechain) [inline consensus-types/blocks/execution.go:653]: This was a bad check. `data` is an interface, and the underlying object could be bellatrix or capella

**Review by rauljordan — APPROVED** [review]

**Review by rauljordan — APPROVED** [review]

File: beacon-chain/core/blocks/payload_test.go [hunk 18]
```diff
@@ -253,7 +253,7 @@ func Test_IsExecutionBlockCapella(t *testing.T) {
 	require.NoError(t, err)
 	got, err := blocks.IsExecutionBlock(wrappedBlock.Body())
 	require.NoError(t, err)
-	require.Equal(t, true, got)
+	require.Equal(t, false, got)
```
Comment (prestonvanloon) [inline beacon-chain/core/blocks/payload_test.go:256]: Why is a Capella block not an execution block?

File: beacon-chain/core/blocks/payload_test.go [hunk 19]
```diff
@@ -349,14 +349,15 @@ func Test_IsExecutionEnabled(t *testing.T) {
...
+
 func Test_IsExecutionEnabledCapella(t *testing.T) {
 	st, _ := util.DeterministicGenesisStateCapella(t, 1)
 	blk := util.NewBeaconBlockCapella()
 	body, err := consensusblocks.NewBeaconBlockBody(blk.Block.Body)
 	require.NoError(t, err)
 	got, err := blocks.IsExecutionEnabled(st, body)
 	require.NoError(t, err)
-	require.Equal(t, true, got)
+	require.Equal(t, false, got)
```
Comment (prestonvanloon) [inline beacon-chain/core/blocks/payload_test.go:360]: Same here, why wouldn't this be enabled in capella?
